src/modules/logger/logger.cpp

```diff
@@ -686,9 +643,56 @@ void Logger::run()
 
 	while (!should_exit()) {
 		// Start/stop logging (depending on logging mode, by default when arming/disarming)
-		const bool logging_started = start_stop_logging();
+		if (should_start_logging()) {
```

Comment:
We also need to initialze if mavlink log is starting, and we should do it only once.